libs/wingsdk/src/utils/resource-names.ts

```diff
@@ -0,0 +1,39 @@
+import { Construct } from "constructs";
+
+export enum CaseConventions {
+  LOWERCASE = "lowercase",
+  UPPERCASE = "uppercase",
+}
+
+export interface NameOptions {
+  readonly regexMatch: RegExp;
```

Comment:
This is currently "disallowedRegEx". I think we should flip the logic and make this "allowedRegEx". It's safer to say which characters are allowed then to try to cover all the disallowed cases (or use a negative regex).

Reply:
I agree, it makes more sense